app/models/manageiq/providers/ibm_cloud/container_manager.rb

```diff
@@ -0,0 +1,201 @@
+class ManageIQ::Providers::IbmCloud::ContainerManager < ManageIQ::Providers::Kubernetes::ContainerManager
+  require_nested :Container
+  require_nested :ContainerGroup
+  require_nested :ContainerNode
+  require_nested :EventCatcher
+  require_nested :EventParser
+  require_nested :Refresher
+  require_nested :RefreshWorker
+
+  def self.ems_type
+    @ems_type ||= "iks".freeze
+  end
+
+  def self.description
+    @description ||= "IBM Cloud Kubernetes Service".freeze
+  end
+
+  def self.display_name(number = 1)
+    n_('Container Provider (IBM Cloud)', 'Container Providers (IBM Cloud)', number)
+  end
+
+  def self.default_port
+    443
+  end
+
+  def connect_options(options = {})
+    authentication = authentication_best_fit(options.fetch(:auth_type, "bearer"))
+    token = self.class.get_token(authentication.auth_key).id_token
+
+    super.merge(
+      :bearer => token,
+      :region => provider_region
+    )
+  rescue IbmCloudIam::ApiError => err
+    error_message = JSON.parse(err.response_body)["message"]
+    _log.error("IAM authentication failed: #{err.code} #{error_message}")
+  end
+
+  def self.kubernetes_auth_options(options)
+    {
+      :bearer_token => options[:bearer]
```

Comment:
@nasark what you want to do is get the token dynamically in this method, rather than in `connect_options` and `verify_credentials` because this is what is re-run each time we connect in the refresh watch threads.  We save the "connect_options" and re-connect with the same options so you'd be saving a stale bearer token.

recommend:
```
:bearer_token => get_token(options[:api_key]).id_token
```

Then passing in the `:api_key` to options in `connect_options` and `verify_credentials`